### DISCUSSION TITLE
Failed to get DNS records for www.youtube.com using server 1.1.1.1

### DISCUSSION THREAD

**Discussion by MaxweIlCatDEV** [Rehike/Rehike#671] [General]
I installed Rehike using the recommended steps from the instructions and I get a YouTube connection error when I try to go to localhost/yotubue.com

> Rehike\Util\Nameserver\DnsLookupException: Failed to get DNS records for www.youtube.com using server 1.1.1.1
  at Rehike\Util\Nameserver\Nameserver::lookup("www.youtube.com", 443)(rehike://modules/Rehike/Util/Nameserver/Nameserver.php:97)
  at Rehike\Util\Nameserver\Nameserver::get("www.youtube.com", "1.1.1.1", 443)(rehike://modules/Rehike/Util/Nameserver/Nameserver.php:62)
  at Rehike\Boot\Tasks::initNetwork()(rehike://modules/Rehike/Boot/Tasks.php:38)
  at Rehike\Boot\Bootloader::runInitTasks()(rehike://modules/Rehike/Boot/Bootloader.php:167)
  at Rehike\Boot\Bootloader::boot()(rehike://modules/Rehike/Boot/Bootloader.php:108)
  at Rehike\Boot\Bootloader::startSession()(rehike://modules/Rehike/Boot/Bootloader.php:24)
  at {main}(rehike://index.php:72)

**Comment by kawapure** [comment]
The Rehike server may not be blocked from outgoing requests by some external force, i.e. network firewall.

**Comment by MaxweIlCatDEV** [comment]
> The Rehike server may not be blocked from outgoing requests by some external force, i.e. network firewall.

I (probably) don't use any firewalls and my network doesn't block anything.

**Comment by kawapure** [comment]
I'll try to debug this with you in something like 12 hours.

**Comment by MaxweIlCatDEV** [comment]
> I'll try to debug this with you in something like 12 hours.

I installed this on my Arch Linux and it works fine. I have no idea what's going wrong on Windows...

**Comment by kawapure** [comment]
Please replace index.php with something like

```php
<?php
$page = file_get_contents("https://www.google.com");
echo $page;
```

to test if PHP is allowed the connect to the internet. If it isn't, then I would investigate Windows Firewall as it might be disallowing outgoing connections from PHP/Apache.

**Comment by MaxweIlCatDEV** [comment]
> to test if PHP is allowed the connect to the internet. If it isn't, then I would investigate Windows Firewall as it might be disallowing outgoing connections from PHP/Apache.

php has access to the internet. I can see the google page.

**Comment by kawapure** [comment]
Possibly 1.1.1.1 is blocked in your country. Could you try using a different DNS server by modifying:

https://github.com/Rehike/Rehike/blob/3af334775ebddf5f1b7766daaf8810d2734b382b/modules/Rehike/Boot/Tasks.php#L38
